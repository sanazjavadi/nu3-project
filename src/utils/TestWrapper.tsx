import { configureStore, Store } from "@reduxjs/toolkit";
import { MemoryRouter } from "react-router-dom";
import { ComponentType, FunctionComponent } from "react";
import { Provider } from "react-redux";
import reducer from "../reducers";

export const makeStore = (): Store => {
  return configureStore({ reducer });
};

const wrapComponent = (
  Component: ComponentType | FunctionComponent | any,
  store: Store | null = null,
  props = {}
): JSX.Element => {
  return (
    <Provider store={store || makeStore()}>
      <MemoryRouter>
        <Component {...props} />
      </MemoryRouter>
    </Provider>
  );
};

export default wrapComponent;
