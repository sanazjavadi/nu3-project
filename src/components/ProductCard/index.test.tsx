/* eslint-disable testing-library/no-node-access */
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import TestWrapper, { makeStore } from "../../utils/TestWrapper";
import ProductCart from "./";

const store = makeStore();
const MockData = [
  {
    price: 13.99,
    image:
      "https://cdn.shopify.com/s/files/1/0095/2274/1329/products/4_252F5_252F3_252F4_252F453437c2922e342acc8db2a47e2a24ee197fe940_20174_9807111a-5f4b-4430-a21b-3809799931aa_large.jpg?v=1623047667",
    vendor: "BEAVITA",
    title: "BEAVITA Vitalkost Diät-Shake",
    id: "4365140131921",
    link: "/products/beavita-vitalkost-diaet-shake",
  },
];
describe("ProductCart Tests Cases", () => {
  test("ProductCart: Renders", () => {
    render(TestWrapper(ProductCart, store, MockData[0]));
    const productCartElement = screen.getByTestId("product-cart-container");
    expect(productCartElement).toBeInTheDocument();
  });

  it("should render a product image", () => {
    render(TestWrapper(ProductCart, store, MockData[0]));
    const productCardImage = screen.getByRole("img");
    expect(productCardImage).toBeInTheDocument();

    expect(productCardImage.getAttribute("src")).toBe(MockData[0].image);
  });

  it("should render a product title", () => {
    render(TestWrapper(ProductCart, store, MockData[0]));
    const productCardTitle = screen.getByText("BEAVITA Vitalkost Diät-Shake");
    expect(productCardTitle).toBeInTheDocument();
  });

  it("should render the product link", () => {
    render(TestWrapper(ProductCart, store, MockData[0]));
    const productCardTitle = screen.getByText("BEAVITA Vitalkost Diät-Shake");
    expect(productCardTitle.closest("a")?.href).toBe(
      "http://localhost/product-detail-page"
    );
  });
});

test("ProductCart: Snapshot", () => {
  const component = renderer.create(TestWrapper(ProductCart, store));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
