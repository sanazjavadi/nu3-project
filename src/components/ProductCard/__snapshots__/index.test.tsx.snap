// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCart: Snapshot 1`] = `
<div
  className="p-4 max-w-sm bg-white rounded-lg border border-gray-200 shadow-sm sm:p-6 lg:p-8 dark:bg-gray-800 dark:border-gray-700 m-2"
  data-testid="product-cart-container"
>
  <img
    alt="product image"
    className="p-8 rounded-t-lg"
  />
  <div
    className="px-5 pb-5"
  >
    <a
      href="/product-detail-page"
      onClick={[Function]}
    >
      <h5
        className="text-lg font-semibold tracking-tight text-gray-900 dark:text-white h-16"
      >
        text
      </h5>
    </a>
    <div
      className="flex justify-between items-center"
    >
      <span
        className="text-3xl font-bold text-gray-900 dark:text-white"
      >
        â‚¬NaN
      </span>
      <button
        className="inline-block px-6 py-2.5 bg-gray-800 text-white font-medium text-xs leading-tight uppercase rounded shadow-md hover:bg-gray-900 hover:shadow-lg focus:bg-gray-900 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-gray-900 active:shadow-lg transition duration-150 ease-in-out"
        data-testid="initial-button"
        onClick={[Function]}
      >
        Add to cart
      </button>
    </div>
  </div>
</div>
`;
